onnx/__init__.py

```diff
@@ -287,6 +287,9 @@ def save_model(
     location: str | None = None,
     size_threshold: int = 1024,
     convert_attribute: bool = False,
+    align_offset: bool = False,
```

Comment:
Should we default this to true?

Reply:
The argument made against this was to keep consistency? This is a behavior change IMO, but not really a breaking behavior change. There is also an increase in model file size with the padding added.

I am in favor of eventually defaulting this on, but I also think ORT needs better support of mapfiles first to save a lot commit memory cost?
See https://github.com/microsoft/onnxruntime/issues/21448

Reply:
I agree we default this to true, I'm working a similar offset alignment PR from ORT: https://github.com/microsoft/onnxruntime/pull/21604 
We want to enable offset alignment when save external data file during inference session initialization by default. With that, there are some optimization script/tool using onnx save_model method to save onnx model (for external initializer as well), for example https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/python/tools/transformers/onnx_model.py#L1164, with align offset default to true in onnx and in ORT, we can have consistent behavior.